qt/widgets/common/src/WorkspacePresenter/WorkspaceTreeWidgetSimple.cpp

```diff
@@ -107,6 +107,25 @@ void WorkspaceTreeWidgetSimple::popupContextMenu() {
       plotSubMenu->addAction(m_overplotSpectrum);
       plotSubMenu->addAction(m_plotSpectrumWithErrs);
       plotSubMenu->addAction(m_overplotSpectrumWithErrs);
+
+      // Don't plot 1D spectra if only one X value
+      bool multipleBins = false;
+      try {
+        multipleBins = (matrixWS->blocksize() > 1);
+      } catch (...) {
+        const size_t numHist = matrixWS->getNumberHistograms();
+        for (size_t i = 0; i < numHist; ++i) {
+          if (matrixWS->y(i).size() > 1) {
+            multipleBins = true;
+            break;
+          }
+        }
+      }
+      m_plotSpectrum->setEnabled(multipleBins);
+      m_overplotSpectrum->setEnabled(multipleBins);
+      m_plotSpectrumWithErrs->setEnabled(multipleBins);
+      m_overplotSpectrumWithErrs->setEnabled(multipleBins);
```

Comment:
Instead of doing `setEnabled` on each action individually, you could do:
```
for (auto action : plotSubMenu->actions()) {
    action->setEnabled(multipleBins);
}
```

This would mean that we don't have to add a line for every new action added in the future.

Reply:
Although that would also disable 2D plotting, which is not a pointless graph, and would be different from Mantidplot.

Reply:
The Colorfill option isn't added to the menu until after this so it wouldn't be affected.